content/en/platform/corda/5.0-beta/developing/ledger/smart-contracts.md

```diff
@@ -38,14 +37,14 @@ Each transaction state specifies a contract type. The contract specified takes t
 The contract code can:
 
 * Check the number of inputs, outputs, commands, or attachments.
-* Check for time windows.
+* Check that a [time window]({{< relref "./time-windows.md" >}}) exists and that time points mentioned in states fall within this time window.
 * Check the contents of all components.
 * Evaluate looping constructs, variable assignments, function calls, helper methods, and other aspects of the transaction code.
 * Group similar states to validate them as a group. For example, it can impose a rule on the combined value of all the cash states.
 
 ## Determinism
 
-For the nodes on a network to reach consensus about a proposed update to the ledger, transaction verification must be deterministic. That means contracts must always accept or always reject a given transaction. For example, a transaction’s validity cannot depend on the time it was validated, or the amount of information the node running the contract holds.
+To enable the nodes on a network to reach consensus about a proposed update to the ledger, transaction verification must be deterministic. This means that contracts must always accept or reject a given transaction, regardless of when it is validated, or the amount of information that the node running the contract holds.
```

Comment:
Hmm - that last sentence is a bit misleading, it's always a reject or an accept, the point being that the result can't change on further validations once it's accepted or rejected.